pinot-common/src/main/java/org/apache/pinot/common/utils/TlsUtils.java

```diff
@@ -447,12 +450,32 @@ static void reloadSslFactoryWhenFileStoreChanges(SSLFactory baseSslFactory,
           LOGGER.info("Detected change in file: {}, try to renew SSLFactory {} "
               + "(built from key store {} and truststore {})",
               changedFile, baseSslFactory, keyStorePath, trustStorePath);
-          SSLFactory updatedSslFactory = createSSLFactory(
-              keyStoreType, keyStorePath, keyStorePassword, trustStoreType, trustStorePath, trustStorePassword,
-              sslContextProtocol, secureRandom, false);
-          SSLFactoryUtils.reload(baseSslFactory, updatedSslFactory);
-          LOGGER.info("Successfully renewed SSLFactory {} (built from key store {} and truststore {}) "
-                  + "on file {} changes", baseSslFactory, keyStorePath, trustStorePath, changedFile);
+          try {
+            // Need to retry a few times because when one file (key store or trust store) is updated, the other file
+            // (trust store or key store) may not have been fully written yet, so we need to wait a bit and retry.
```

Comment:
Please also add a note that it's ok for 3 retries failure, the watcher will be triggered again and renew.